account_statement_import_sheet_file/models/account_statement_import.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 CorporateHub (https://corporatehub.eu)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountStatementImport(models.TransientModel):
+    _inherit = "account.statement.import"
+
+    def _get_default_mapping_id(self):
+        return (
+            self.env["account.journal"]
+            .browse(self.env.context.get("journal_id"))
+            .default_sheet_mapping_id
+        )
+
+    sheet_mapping_id = fields.Many2one(
+        string="Sheet mapping",
+        comodel_name="account.statement.import.sheet.mapping",
+        default=_get_default_mapping_id,
+    )
+
+    def _parse_file(self, data_file):
+        self.ensure_one()
+        if self.sheet_mapping_id:
+            try:
+                Parser = self.env["account.statement.import.sheet.parser"]
+                return Parser.parse(
+                    data_file, self.sheet_mapping_id, self.statement_filename
+                )
+            except BaseException as exc:
+                if self.env.context.get("account_statement_import_sheet_file_test"):
+                    raise
+                _logger.warning("Sheet parser error", exc_info=True)
+                raise UserError(_("Bad file/mapping: ") + str(exc)) from exc
```

Comment:
Non-blocking comment, but if you can update the translation method in all occurrences, that would be great.

```suggestion
                raise UserError(self.env._("Bad file/mapping: ") + str(exc)) from exc
```